infrastructure/ansible/roles/flyway/tasks/main.yml

```diff
@@ -36,13 +36,25 @@
   shell: echo "alter role {{ flyway_db_user }} with password '{{ flyway_db_password }}';"  | sudo -u postgres psql
   become_user: postgres
 
+- name: check for flyway download 
+  stat:
+    path: /tmp/flyway-cli.tar.gz
+  register: flyway_download
+
 - name: Download flyway CLI
+  when: flyway_download.stat.exists == False
```

Comment:
Also the default behaviour of the `get_url` module is not to overwrite any existing files according to the docs:
See parameter `force`:
https://docs.ansible.com/ansible/latest/modules/get_url_module.html

Reply:
This update is required, or you get this error:
 "msg": "Destination /tmp/flyway-cli.tar.gz is not writable",

`force` is not ideal as we get a DL and overwrite every time. It turns out if we set a checksum on the get_url task, then it knows enough when to skip download task. I think it's designed this way to have you clean up downloaded files, and/or overwrite them, or set a checksum for them. In any of these cases, if the download fails or is corrupted, then the playbook can be repeated and it'll repair any issues. If you skip download with a corrupt file, then the re-run would not fix the issues.